### PR TITLE
Move "public" resources into R4 Project

### PR DESCRIPTION
**Pull request by mattwiller**
Creates a special Project for holding the base R4 FHIR specification conformance resources.  The goal is to allow inclusion of those resources into other projects effectively via reference, so the hundreds of `StructureDefinition` and other resources do not live in each Project separately.  This change includes:
- Creation of an R4 project with a well-known ID (the V5 UUID generated for name "R4" under the [Nil UUID](https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.7) namespace
- A database migration to move all "orphan" resources into the R4 project at server startup
- Updates to the seed script to create spec definition resources in the R4 project at first server startup
- Login in the `Repository` to add the R4 `Project` compartment to all DB read queries; blocking all writes across project boundaries

Related to #2581

### CODE REVIEW

**Comment by reshmakh** [comment]
Discussed - test in staging.  @mattwiller @codyebberson

File: packages/server/src/fhir/operations/export.ts [hunk 1]
```diff
@@ -113,12 +113,14 @@ function getResourceTypesByExportLevel(exportLevel: string): ResourceType[] {
   return getResourceTypes();
 }
 
+const unexportedResourceTypes = ['CodeSystem', 'SearchParameter', 'StructureDefinition', 'ValueSet'];
+
 function canBeExported(resourceType: string): boolean {
   if (resourceType === 'BulkDataExport') {
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/export.ts:119]: Now that we're introducing `unexportedResourceTypes`, should we move the `BulkDataExport` into that?

Then this method becomes `return !protectedResourcesTypes.includes(resourceType) && !unexportedResourceTypes.includes(resourceType)`

File: packages/server/src/migrations/v48.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  });
+  await moveOrphanResourcesIntoProject('StructureDefinition', r4ProjectId, client);
+  await moveOrphanResourcesIntoProject('SearchParameter', r4ProjectId, client);
+  await moveOrphanResourcesIntoProject('ValueSet', r4ProjectId, client);
+}
+
+const updateQuery = `UPDATE "__TABLE__" SET
+    "projectId" = $1::UUID,
+    compartments = ARRAY[$1]::UUID[],
+    content = jsonb_set(content::jsonb, '{meta, project}'::text[], to_jsonb($1), true)
```
Comment (codyebberson) [inline packages/server/src/migrations/v48.ts:25]: Clever

**Review by codyebberson — APPROVED** [review]
Nice, this looks good.

I'll deploy to staging when I get a chance.  Assuming that looks good, then this is good-to-go 👍

File: packages/server/src/migrations/v48.ts (hunk 2 above)
Comment (mattwiller) [reply packages/server/src/migrations/v48.ts:25]: 😅

**Comment by codyebberson** [comment]
Staging deploy failed:

image

Does the migration handle the case where the `Project` is deleted, and the `content` column is empty?

File: packages/server/src/migrations/v48.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  WHERE
+    "projectId" IS NULL OR "projectId" = $1`;
```
Comment (codyebberson) [inline packages/server/src/migrations/v48.ts:27]: Might need to add `AND "content"<>''`

**Review by codyebberson — APPROVED** [review]

**Pull request merged**